packages/language-server/src/edit/handlers/moveXHandler.ts

```diff
@@ -0,0 +1,40 @@
+import { IncrementalUpdate, MoveEdit } from "@hylimo/diagram-protocol";
+import { EditHandler } from "./editHandler.js";
+import { AbsolutePoint, DefaultEditTypes, RelativePoint } from "@hylimo/diagram-common";
+import { roundToPrecision } from "../../util/roundToPrecision.js";
+
+/**
+ * Handler for moveX edits
+ */
+export const moveXHandler: EditHandler<MoveEdit> = {
+    type: DefaultEditTypes.MOVE_X,
+
+    predictActionDiff(layoutedDiagram, lastApplied, newest, elements) {
+        const deltaX = newest.dx - (lastApplied?.dx ?? 0);
+        const updates: IncrementalUpdate[] = [];
+        for (const element of elements) {
+            if (AbsolutePoint.isAbsolutePoint(element)) {
+                element.x += deltaX;
+                updates.push({
+                    target: element.id,
+                    changes: {
+                        x: element.x
+                    }
+                });
+            } else if (RelativePoint.isRelativePoint(element)) {
```

Comment:
it can be a CanvasElement
in this case the first full update _should_ add a new absolute point to the diagram and set its coordinates accordingly
this will stop predictions, as the topology of the diagram has changed
however, note that before the result of said full update is there, prediction is still called. As there is no way to set the position for that canvas element, it is just ignored (while we could abuse the relative offset, there's really no point for the 20 frames or so before the full update result stops predictions anyways)

Reply:
it should not be a line point, as move x/y edits are not valid for line points
`unset` is a whole different style system concept (and if you mean just no point set, that's the CanvasElement case I described above)